frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileWriter.java

```diff
@@ -0,0 +1,66 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import io.vavr.collection.List;
+import lombok.CustomLog;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.qual.guieffect.IgnoreUIThreadUnsafeCalls;
+
+@CustomLog
+public final class MacheteFileWriter {
+
+  private MacheteFileWriter() {}
+
+  /**
+   * Method for writing branch layout using IntelliJ's VFS API, should executed on UI thread and wrapped in a WriteAction
+   * @param path a path to file where branch layout should be written
+   * @param branchLayout layout to be written
+   * @param backupOldFile flag stating if old layout file should be backed-up
+   * @param requestor object requesting write execution (can be used for debugging when listening for file changes)
+   */
+  @IgnoreUIThreadUnsafeCalls({"java.io.BufferedOutputStream.<init>(java.io.OutputStream)",
+      "java.io.BufferedOutputStream.close()"})
+  @UIEffect
+  public static void writeBranchLayout(Path path, BranchLayout branchLayout, boolean backupOldFile,
+      IBranchLayoutWriter branchLayoutWriter, @Nullable Object requestor)
```

Comment:
Hmmm I'd reorder the params a bit... having `branchLayoutWriter` after `branchLayout` feels odd